src/Codeception/Module/Db.php

```diff
@@ -205,156 +194,271 @@ class Db extends CodeceptionModule implements DbInterface
     ];
 
-    /**
-     * @var bool
-     */
-    protected $populated = false;
-
-    /**
-     * @var \Codeception\Lib\Driver\Db
-     */
-    public $driver;
-
     /**
      * @var array
      */
+    protected $requiredFields = ['dsn', 'user', 'password'];
+    const DEFAULT_DATABASE = 'default';
+
+    public $drivers = [];
+    public $dbhs = [];
+    public $databasesPopulated = [];
+    public $databasesSql = [];
     protected $insertedRows = [];
+    public $currentDatabase = self::DEFAULT_DATABASE;
 
-    /**
-     * @var array
-     */
-    protected $requiredFields = ['dsn', 'user', 'password'];
+    public function getDatabases()
+    {
+        $databases = [$this->currentDatabase => $this->config];
+
+        if (!empty($this->config['databases'])) {
+            foreach ($this->config['databases'] as $databaseKey => $databaseConfig) {
+                $databases[$databaseKey] = array_merge([
+                    'populate' => false,
+                    'cleanup' => false,
+                    'reconnect' => false,
+                    'dump' => null,
+                    'populator' => null,
+                ], $databaseConfig);
+            }
+        }
+        return $databases;
+    }
+    public function connectToDatabases()
+    {
+        foreach ($this->getDatabases() as $databaseKey => $databaseConfig) {
+            $this->connect($databaseKey, $databaseConfig);
+        }
+    }
+    public function cleanUpDatabases()
+    {
+        foreach ($this->getDatabases() as $databaseKey => $databaseConfig) {
+            $this->_cleanup($databaseKey, $databaseConfig);
+        }
+    }
+    public function populateDatabases($configKey)
+    {
+        foreach ($this->getDatabases() as $databaseKey => $databaseConfig) {
+            if ($databaseConfig[$configKey]) {
+                $this->_loadDump($databaseKey, $databaseConfig);
+            }
+        }
+    }
+    public function readSqlForDatabases()
+    {
+        foreach ($this->getDatabases() as $databaseKey => $databaseConfig) {
+            $this->readSql($databaseKey, $databaseConfig);
+        }
+    }
+    public function removeInsertedForDatabases()
+    {
+        foreach ($this->getDatabases() as $databaseKey => $databaseConfig) {
+            $this->amConnectedToDatabase($databaseKey);
+            $this->removeInserted($databaseKey, $databaseConfig);
+        }
+    }
+    public function disconnectDatabases()
+    {
+        foreach ($this->getDatabases() as $databaseKey => $databaseConfig) {
+            $this->disconnect($databaseKey, $databaseConfig);
+        }
+    }
+    public function reconnectDatabases()
+    {
+        foreach ($this->getDatabases() as $databaseKey => $databaseConfig) {
+            if ($databaseConfig['reconnect']) {
+                $this->disconnect($databaseKey, $databaseConfig);
+                $this->connect($databaseKey, $databaseConfig);
+            }
+        }
+    }
+
+    public function __get($name)
```

Comment:
Ok. We don't need this hook to be used on long-term
Please add a deprecation warning here. 
`Codeception\Lib\Notification::deprecate`

Reply:
Fixed